File: src/main/java/com/canefe/story/NPCData.java
package com.canefe.story;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

// this is how we save NPC data AI
public class NPCData {
    private final String name;
    private final String role; // why?
    private final StoryLocation storyLocation; // location of the NPC for local general context
    private final String context;
    private final List<Story.ConversationMessage> conversationHistory; // conversation history of the NPC

    public NPCData(String name, String role, StoryLocation storyLocation, String context, List<Story.ConversationMessage> conversationHistory) {
        this.name = name;
        this.role = role;
        this.storyLocation = storyLocation;
        this.context = context;
        this.conversationHistory = conversationHistory;
    }

    public String getName() {
        return name;
    }

    public String getRole() {
        return role;
    }

    public StoryLocation getLocation() {
        return storyLocation;
    }

    public String getContext() {
        return context;
    }

    public List<Story.ConversationMessage> getConversationHistory() {
        return new ArrayList<>(conversationHistory);
    }

    public String toString() {
        return "NPCData{name=" + name + ", role=" + role + ", location=" + storyLocation + ", context=" + context + "}";
    }


}

File: src/main/kotlin/com/canefe/story/npc/data/NPCData.kt
package com.canefe.story.npc.data

import com.canefe.story.StoryLocation
import com.canefe.story.conversation.ConversationMessage

data class NPCData(
    val name: String,
    val role: String,
    val storyLocation: StoryLocation,
    val context: String,
    private val conversationHistory: List<ConversationMessage>
) {
    val avatar: String = ""

    fun getConversationHistory(): List<ConversationMessage> {
        return conversationHistory.toList() // Defensive copy
    }

    override fun toString(): String {
        return "NPCData{name=$name, role=$role, location=$storyLocation, context=$context}"
    }
}
